Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -872,6 +892,11 @@ bool CStaticFunctionDefinitions::IsElementLowLod(CElement* pElement, bool& bOutI
             bOutIsLowLod = pObject->IsLowLod();
             break;
         }
+        case CElement::BUILDING:
+        {
+            bOutIsLowLod = static_cast<CBuilding*>(pElement)->GetHighLodObject() ? true : false;
```

Comment:
> `static_cast() != nullptr`?

`static_cast` is compile time cast

Reply:
> And?

`static_cast` cannot return `nullptr` in runtime.

Reply:
1. of course it can (https://godbolt.org/z/186YoYP3c)
2. doesnt matter because the end object is not static_casted

Reply:
It cannot be `nullptr` as I already said. Because static_cast doesn't exist in runtime.

Reply:
Sure. But in this particular place pElement will always be valid. In addition to that it will always be CBuilding. And consequently the result of static_cast will always be not-nullptr.

Reply:
Great but thats irrelevant to what I said

Reply:
> Great but thats irrelevant to what I said

I would say that the words you said in last three comments was irrelevant to my original comment.

Reply:
Thanks for the enlightenment but comments under PR is not a place for this.

Reply:
...its... part of the conversation...

Reply:
I see you have a need for demonstrating your knowledge. But in this particular conversation this is redundant. Because my comments were in the context of a concrete line of code. However, you are trying to expand the discussion to common principles steering the conversation away from the original question. This is what is called offtopic and I would love for you to avoid this practice.